src/aepp-wallet-communication/connection/BrowserRuntime.ts

```diff
@@ -34,19 +37,27 @@ export default class BrowserRuntimeConnection extends BrowserConnection {
   override disconnect(): void {
     super.disconnect();
     this.port.disconnect();
+    if (this.#listeners == null) throw new UnexpectedTsError();
+    this.port.onMessage.removeListener(this.#listeners[0]);
+    this.port.onDisconnect.removeListener(this.#listeners[1]);
+    this.#listeners = undefined;
   }
 
   override connect(
     onMessage: (message: any, origin: string, source: Runtime.Port) => void,
     onDisconnect: () => void,
   ): void {
     super.connect(onMessage, onDisconnect);
-    this.port.onMessage.addListener((message, port) => {
-      this.receiveMessage(message);
-      // TODO: make `origin` optional because sender url is not available on aepp side
-      onMessage(message, port.sender?.url ?? '', port);
-    });
-    this.port.onDisconnect.addListener(onDisconnect);
+    this.#listeners = [
+      (message, port) => {
+        this.receiveMessage(message);
+        // TODO: make `origin` optional because sender url is not available on aepp side
+        onMessage(message, port.sender?.url ?? '', port);
+      },
+      onDisconnect,
+    ];
+    this.port.onMessage.addListener(this.#listeners[0]);
+    this.port.onDisconnect.addListener(this.#listeners[1]);
```

Comment:
This needs to be refactored, this code actually binds listeners, not connecting no anything